src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.netfx.cs

```diff
@@ -105,7 +103,11 @@ private static void AddRequestTagsAndInstrumentRequest(HttpWebRequest request, A
                 activity.AddTag(SpanAttributeConstants.HttpMethodKey, request.Method);
                 activity.AddTag(SpanAttributeConstants.HttpHostKey, HttpTagHelper.GetHostTagValueFromRequestUri(request.RequestUri));
                 activity.AddTag(SpanAttributeConstants.HttpUrlKey, request.RequestUri.OriginalString);
-                activity.AddTag(SpanAttributeConstants.HttpFlavorKey, HttpTagHelper.GetFlavorTagValueFromProtocolVersion(request.ProtocolVersion));
+
+                if (Options.SetHttpFlavor == true)
```

Comment:
Do we need a generic callback here to collect additional attributes?

Reply:
Seems like it would be useful. After this PR I want to extend the options. I was thinking I would do an issue first with the proposed API so we can discuss, I'll add that on there?